### PR TITLE
Sarkars/backend create tensor

### PR DESCRIPTION
**Pull request by sayantan-nervana**
For certain backends (even if their executable can create tensors), use backends to create tensors

### CODE REVIEW

File: test/tf_exec.cpp [hunk 1]
```diff
@@ -138,6 +138,36 @@ TEST(tf_exec, SingleGraphOn2Threads) {
   }
 }
 
+TEST(tf_exec, model) {
```
Comment (sayantan-nervana) [inline test/tf_exec.cpp]: TODO remove this

File: test/tf_exec.cpp [hunk 2]
```diff
@@ -95,7 +95,16 @@ Status CreateSession(const string& graph_filename, const string& backend_name,
...
+// This test might fail when running with persistent output tensors
+// Maybe because once we have computed outputs out to persistent tensors,
+// the next thread comes in before Compare runs, and changes the values?
+// For example, if we add a 1sec sleep right after entering Compute(), this test
+// would pass (since we now allow enough time for compare to run before the next
+// thread comes in and modifies the persistent tensor values)
+// TODO: see how persistenttensors might fit in with this kind of multithreading
+// (symmetric parallel)
 TEST(tf_exec, SingleGraphOn2Threads) {
+  SetEnvVariable("NGRAPH_TF_NOT_PERSISTENT", "1");
```
Comment (sayantan-nervana) [inline test/tf_exec.cpp]: `NOT`->`DISABLE`

File: ngraph_bridge/ngraph_encapsulate_impl.h [hunk 3]
```diff
@@ -174,6 +174,18 @@ class NGraphEncapsulateImpl {
...
   }
 
+  void ClearNgExecPersistentOutputCache() { m_out_persistents.clear(); }
+
+  Status GetPersistentTFOutputTensor(
+      std::shared_ptr<ngraph::runtime::Executable>,
+      std::vector<tensorflow::PersistentTensor>&);
+
+  bool PersistentOutputsExist(std::shared_ptr<ngraph::runtime::Executable>);
+
```
Comment (kanvi-nervana) [inline ngraph_bridge/ngraph_encapsulate_impl.h]: Can you add some comments explaining what this function is doing?

File: ngraph_bridge/ngraph_encapsulate_impl.h [hunk 4]
```diff
@@ -174,6 +174,18 @@ class NGraphEncapsulateImpl {
...
+  Status RegisterOutputPersistentTensors(
```
Comment (kanvi-nervana) [inline ngraph_bridge/ngraph_encapsulate_impl.h]: RegisterOutputPersistentTensors  ----> RegisterPersistentOutputTensors

File: ngraph_bridge/ngraph_encapsulate_impl.cc [hunk 5]
```diff
@@ -576,6 +580,36 @@ void NGraphEncapsulateImpl::DumpNgFunction(
   StringToFile(file_name, m_serialized_ng_function_map[ng_exec]);
 }
 
+Status NGraphEncapsulateImpl::GetPersistentTFOutputTensor(
+    std::shared_ptr<ngraph::runtime::Executable> exec,
+    std::vector<tensorflow::PersistentTensor>& tf_output_tensors) {
+  auto itr = m_out_persistents.find(exec);
+  if (itr == m_out_persistents.end()) {
```
Comment (shresthamalik) [inline ngraph_bridge/ngraph_encapsulate_impl.cc]: Could use PersistentOutputsExist

File: ngraph_bridge/ngraph_encapsulate_impl.cc [hunk 6]
```diff
@@ -576,6 +580,36 @@ void NGraphEncapsulateImpl::DumpNgFunction(
...
+bool NGraphEncapsulateImpl::PersistentOutputsExist(
+    std::shared_ptr<ngraph::runtime::Executable> exec) {
+  return m_out_persistents.find(exec) != m_out_persistents.end();
+}
+
+Status NGraphEncapsulateImpl::RegisterPersistentOutputTensors(
+    std::shared_ptr<ngraph::runtime::Executable> exec,
+    std::vector<tensorflow::PersistentTensor> persistent_tensors) {
+  auto itr = m_out_persistents.find(exec);
+  if (itr != m_out_persistents.end()) {
```
Comment (shresthamalik) [inline ngraph_bridge/ngraph_encapsulate_impl.cc]: Could use PersistentOutputsExist

**Comment by shresthamalik** [comment]
Please look into the failing pipelines

File: ngraph_bridge/ngraph_encapsulate_impl.cc (hunk 5 above)
Comment (sayantan-nervana) [reply ngraph_bridge/ngraph_encapsulate_impl.cc]: Then I'd be doing 2 searches. One in `PersistentOutputsExist` and one in again to actually return the item. Right now I do only one search to find the appropriate iterator

**Comment by sayantan-nervana** [comment]
> Please look into the failing pipelines
The usual suspects pipelines are failing with the "gcc7" and "whl not found errors"

**Pull request merged**